misc/python/materialize/scalability/operation.py

```diff
@@ -35,20 +35,65 @@ def _execute(self, data: OperationData) -> OperationData:
         raise NotImplementedError
 
 
-class SqlOperation(Operation):
+class SqlOperationWithInput(Operation):
     def required_keys(self) -> set[str]:
-        return {"cursor"}
+        return {"cursor"}.union(self.required_input_keys())
+
+    def required_input_keys(self) -> set[str]:
+        raise NotImplementedError
 
     def _execute(self, data: OperationData) -> OperationData:
         try:
             cursor: Cursor = data.cursor()
-            cursor.execute(self.sql_statement().encode("utf8"))
+            cursor.execute(self.sql_statement_based_on_input(data).encode("utf8"))
             cursor.fetchall()
         except ProgrammingError as e:
             assert "the last operation didn't produce a result" in str(e)
 
         return data
 
+    def sql_statement_based_on_input(self, input: OperationData) -> str:
+        raise NotImplementedError
+
+
+class SqlOperationWithSeed(SqlOperationWithInput):
+    def __init__(self, seed_key: str):
+        self.seed_key = seed_key
```

Comment:
Is seed the right term here? Seems like it is actually a `parameter` . So `SqlOperationWithParameter` , etc?